dex/farm-staking-proxy/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+
+        for p in &payments {
+            let attributes = self.get_dual_yield_token_attributes(p.token_nonce)?;
+            let staking_farm_token_amount =
+                self.get_staking_farm_token_amount_equivalent(&attributes, &p.amount);
+
+            staking_farm_tokens.push(EsdtTokenPayment::new(
+                staking_farm_token_id.clone(),
+                attributes.staking_farm_token_nonce,
+                staking_farm_token_amount,
+            ));
+
+            let lp_farm_token_amount =
+                self.get_lp_farm_token_amount_equivalent(&attributes, &p.amount);
+            let lp_tokens_in_position = self.get_lp_tokens_in_farm_position(
+                attributes.lp_farm_token_nonce,
+                &attributes.lp_farm_token_amount,
+            )?;
+            let new_staking_farm_value =
+                self.get_lp_tokens_value_in_staking_token(&lp_tokens_in_position);
+
+            lp_farm_tokens.push(EsdtTokenPayment::new(
+                lp_farm_token_id.clone(),
+                attributes.lp_farm_token_nonce,
+                lp_farm_token_amount,
+            ));
+            new_staking_farm_values.push(new_staking_farm_value);
+        }
+
+        let lp_farm_address = self.lp_farm_address().get();
+        let lp_farm_result: ClaimRewardsResultType<Self::Api> = self
+            .lp_farm_proxy_obj(lp_farm_address)
+            .claim_rewards(OptionalArg::None)
+            .with_multi_token_transfer(lp_farm_tokens)
+            .execute_on_dest_context();
+        let (new_lp_farm_tokens, lp_farm_rewards) = lp_farm_result.into_tuple();
+
+        let staking_farm_address = self.staking_farm_address().get();
+        let staking_farm_result: ClaimRewardsResultType<Self::Api> = self
+            .staking_farm_proxy_obj(staking_farm_address)
+            .claim_rewards_with_new_value(staking_farm_tokens, new_staking_farm_values)
+            .execute_on_dest_context();
+        let (new_staking_farm_tokens, staking_farm_rewards) = staking_farm_result.into_tuple();
+
+        let caller = self.blockchain().get_caller();
+        self.send().direct(
+            &caller,
+            &lp_farm_rewards.token_identifier,
+            lp_farm_rewards.token_nonce,
+            &lp_farm_rewards.amount,
+            &[],
+        );
+        self.send().direct(
+            &caller,
+            &staking_farm_rewards.token_identifier,
+            staking_farm_rewards.token_nonce,
+            &staking_farm_rewards.amount,
+            &[],
+        );
+        self.create_and_send_dual_yield_tokens(
+            &caller,
+            &new_staking_farm_tokens.amount,
+            new_lp_farm_tokens.token_nonce,
+            new_lp_farm_tokens.amount,
+            new_staking_farm_tokens.token_nonce,
+            new_staking_farm_tokens.amount.clone(),
+        );
+
+        Ok(())
+    }
+
+    #[payable("*")]
+    #[endpoint(unstakeFarmTokens)]
+    fn unstake_farm_tokens(
```

Comment:
this function is too big - it could be refactored in 3 small ones - exit_farm, remove_liquidity, and unstake

Reply:
done